relay-server/src/actors/processor.rs

```diff
@@ -2753,22 +2753,14 @@ impl Service for EnvelopeProcessorService {
         tokio::spawn(async move {
             let semaphore = Arc::new(Semaphore::new(thread_count));
 
-            let Ok(mut subscription) = self.inner.global_config.send(Subscribe).await else {
-                // TODO(iker): we accept this sub-optimal error handling. TBD
-                // the approach to deal with failures on the subscription
-                // mechanism.
-                relay_log::error!("failed to subscribe to GlobalConfigService");
-                return;
-            };
-
             loop {
                 let next_msg = async { tokio::join!(rx.recv(), semaphore.clone().acquire_owned()) };
 
                 tokio::select! {
                    biased;
 
                     // TODO(iker): deal with the error when the sender of the channel is dropped.
-                    _ = subscription.changed() => self.global_config = subscription.borrow().clone(),
+                    _ = self.global_config_receiver.changed() => self.global_config = self.global_config_receiver.borrow().clone(),
                     (Some(message), Ok(permit)) = next_msg => {
                         let service = self.clone();
```

Comment:
i was a bit worried about the cloning of the receiver, but according to the source code it seems trivial.  
 
we could avoid it, perhaps by making a broker with the global_config and the `inner` field, and moving all the methods to it. this could then be cloned separately from the receiver.